### ISSUE TITLE
Guidance on Selecting Data for LightGCN

### ISSUE THREAD

**Issue by MRahimii** [kuandeng/LightGCN#25]
I have encountered some difficulties in selecting the exact data used in your experiments from the original Yelp dataset. Specifically, I have noticed that the list of users and items provided on GitHub does not completely match the original Yelp dataset I have accessed.

Could you kindly provide guidance on how to accurately select the data used in your study? Additionally, if possible, could you share the original dataset or any supplementary materials that would assist in this endeavor?

Thank you very much for your time and assistance

**Comment by atesnrllh** [comment]
Hi, thanks for raising this issue — I also ran into a related problem.

While examining the TensorFlow implementation (specifically the `Data` class), I noticed that the code assumes a very specific structure of `train.txt` and `test.txt`:

**1. Every user appearing in the test file must also appear in the train file**, because:

```python
self.train_items[u]
```

is accessed during negative sampling inside `sample_test()`.  
If a test user does not appear in `train.txt`, this will trigger a `KeyError`.

**2. All item IDs in the test set must fall within the global item ID range** (defined by train + test), otherwise the item embedding table is not initialized properly.

Because this repository does *not* perform any internal train/test split, the provided `train.txt` and `test.txt` must have been generated by an external preprocessing pipeline.

So my question is:

**Could you clarify the exact preprocessing procedure used to generate these files?**

Specifically:
- How was the leave-one-out split applied?
- How did you ensure that every test user appears in the training set (which is necessary for the current implementation to run without errors)?

Understanding this pipeline would help users prepare custom datasets that are compatible with this implementation.

Thank you!
